service/pkg/config/environment_value_loader.go

```diff
@@ -56,7 +59,41 @@ func (l *EnvironmentValueLoader) Get(key string) (any, error) {
 
 // GetConfigKeys returns all the configuration keys found in the environment variables.
 func (l *EnvironmentValueLoader) GetConfigKeys() ([]string, error) {
-	return l.viper.AllKeys(), nil
+	// Start with keys Viper already knows about (from defaults/file loads)
+	keys := make([]string, 0, len(l.viper.AllKeys()))
+	keys = append(keys, l.viper.AllKeys()...)
+
+	// Discover environment variables with the configured prefix and convert them to dotted keys
+	// Example: OPENTDF_SERVICES_KAS_ROOT_KEY -> services.kas.root_key
+	// Note: Viper treats keys case-insensitively; we normalize to lower-case dotted form here
+	prefix := l.envPrefix + "_"
+	for _, env := range os.Environ() {
```

Comment:
This bit looks like what Viper should be doing already. If it is adding additional keys we should cache those keys and values so they can be returned by `Get`, otherwise we might hit this: https://github.com/opentdf/platform/blob/main/service/pkg/config/config.go#L219-L229